libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigationProvider.kt

```diff
@@ -33,6 +39,10 @@ object MapboxNavigationProvider {
      * @see [isCreated]
      */
     @JvmStatic
+    @Deprecated(
+        message = "Get the MapboxNavigation instance through MapboxNavigationObserver or" +
+            " MapboxNavigationApp.current"
+    )
```

Comment:
Let's provide `ReplaceWith` annotation to allow IDEs to automatically apply the replacements. ([here an example](
https://github.com/mapbox/mapbox-navigation-android/blob/47e17155e6766a84b6de012d8e6874d65378ffcd/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/data/MapboxNavigationViewportDataSource.kt#L395))

Reply:
This one could potentially have a replace with. Using `MapboxNavigationApp.current`

But the actual recommendation is to use `MapboxNavigationApp.registerObserver` so I didn't add it